pkg/kv/kvserver/store_snapshot.go

```diff
@@ -829,7 +856,56 @@ func (kvSS *kvBatchSnapshotStrategy) Send(
 	}
 
 	var valBuf []byte
-	if sharedReplicate {
+	if sharedReplicate || externalReplicate {
+		var sharedVisitor func(sst *pebble.SharedSSTMeta) error
+		if sharedReplicate {
+			sharedVisitor = func(sst *pebble.SharedSSTMeta) error {
+				sharedSSTCount++
+				snap.sharedBackings = append(snap.sharedBackings, sst.Backing)
+				backing, err := sst.Backing.Get()
+				if err != nil {
+					return err
+				}
+				ikeyToPb := func(ik pebble.InternalKey) *kvserverpb.SnapshotRequest_SharedTable_InternalKey {
+					return &kvserverpb.SnapshotRequest_SharedTable_InternalKey{
+						UserKey: ik.UserKey,
+						Trailer: ik.Trailer,
+					}
+				}
+				sharedSSTs = append(sharedSSTs, kvserverpb.SnapshotRequest_SharedTable{
+					Backing:          backing,
+					Smallest:         ikeyToPb(sst.Smallest),
+					Largest:          ikeyToPb(sst.Largest),
+					SmallestRangeKey: ikeyToPb(sst.SmallestRangeKey),
+					LargestRangeKey:  ikeyToPb(sst.LargestRangeKey),
+					SmallestPointKey: ikeyToPb(sst.SmallestPointKey),
+					LargestPointKey:  ikeyToPb(sst.LargestPointKey),
+					Level:            int32(sst.Level),
+					Size_:            sst.Size,
+				})
+				return nil
+			}
+		}
+		var externalVisitor func(sst *pebble.ExternalFile) error
+		if externalReplicate {
+			externalVisitor = func(sst *pebble.ExternalFile) error {
+				externalSSTCount++
+				externalSSTs = append(externalSSTs, kvserverpb.SnapshotRequest_ExternalTable{
+					Locator:         []byte(sst.Locator),
+					ObjectName:      sst.ObjName,
+					Size_:           sst.Size,
+					StartKey:        sst.Bounds.Start,
+					EndKey:          sst.Bounds.End,
```

Comment:
Let's merge this after a rebase has happened on a pebble version bump so we can pick up our end bound inclusive bool.

Reply:
Doneburger.